### DISCUSSION TITLE
Variables from context gone after using redis

### DISCUSSION THREAD

**Discussion by fr6nco** [redpanda-data/connect#1702] [Q&A] [answered]
Hi, I need to work with a variable with the redis module multiple times. Im using KeyDB instead of redis since I need to set TTL on subkeys in a hashset. My pipelines are the following

```
    - label: grokparse
      grok:
        expressions:
          - "%{IPV4:ip}.*"
    - label: debug1
      log:
        message: logwhat
        fields_mapping: |
          root.ip = this.ip
    - label: redishset
      redis:
        url: redis://localhost:6379
        command: HSET
        args_mapping: root = [ "hello", this.ip, "world" ]
    - label: redisexpire
      redis:
        url: redis://localhost:6379
        command: EXPIREMEMBER
        args_mapping: root = [ "hello", this.ip, 10 ]
    - label: debug
      log:
        message: log
        fields_mapping: |
          root.ip = this.ip
```

To set args for the command we have to set the the args_mapping's root param. Not sure if this context is pipeline step specific or global, but with the first redis command I'm actually losing the var this.ip in the second step.

```
    INFO logwhat                                       @service=benthos ip=194.108.128.215 label=debug1 path=root.pipeline.processors.4
    INFO log                                           @service=benthos ip="<nil>" label=debug path=root.pipeline.processors.7
```

So basically is I use redis in any step I lose the content in my this context.
Is there any way to overcome this limitation?

**Comment by mihaitodor** [comment, marked as answer]
Hey @fr6nco, the [`redis`](https://www.benthos.dev/docs/components/processors/redis) processor replaces the contents of the input message with the result of the operation. As mentioned in the documentation that I linked, you can work around that by wrapping your `redis` processors in [`branch` processors](https://www.benthos.dev/docs/components/processors/branch). Hope that helps.
